crypto/bio/bio_addr.c

```diff
@@ -747,8 +759,22 @@ int BIO_lookup_ex(const char *host, const char *service, int lookup_type,
                 goto retry;
             }
 # endif
+# if !defined(_WIN32_WCE)
             ERR_raise_data(ERR_LIB_BIO, ERR_R_SYS_LIB,
                            gai_strerror(old_ret ? old_ret : gai_ret));
+# else
+            {
+                wchar_t *wstr_error = gai_strerrorW(old_ret ? old_ret : gai_ret);
+                size_t length = wcslen(wstr_error) + 1;
+                char *str_error = OPENSSL_malloc(length);
+
+                if (str_error != NULL) {
+                    wcstombs(str_error, wstr_error, length);
+                    ERR_raise_data(ERR_LIB_BIO, ERR_R_SYS_LIB, str_error);
```

Comment:
Likewise (also consider factoring this duplicate code into a function)

Reply:
I would strongly suggest writing an `OSSL_ gai_strerror()` that has the correct behaviour in all cases, rather than in inline `#if-#else-#endif`